Feature/EPMUII-10417 Configure e2e tests on playwright

rrodionov91 opened the pull request:
- created folder and docker config for e2e tests
- added serve for prod build
- configured docker for app and tests run
- added github workflow for e2e tests

rrodionov91 committed:
EPMUII-10417:
- created folder and docker config for e2e tests

rrodionov91 committed:
EPMUII-10417:
- added serve for prod build
- configured docker for app and tests run
- added github workflow for e2e tests

rrodionov91 committed:
EPMUII-10417:
- added github workflow for e2e tests

rrodionov91 committed:
EPMUII-10417:
- changed node version on gh workflow

rrodionov91 committed:
EPMUII-10417:
- test gh workflow

rrodionov91 committed:
EPMUII-10417:
- test gh workflow

rrodionov91 committed:
EPMUII-10417:
- test gh workflow

rrodionov91 committed:
EPMUII-10417:
- test gh workflow

rrodionov91 committed:
EPMUII-10417:
- test gh workflow

rrodionov91 committed:
EPMUII-10417:
- test gh workflow

rrodionov91 committed:
EPMUII-10417:
- test gh workflow

rrodionov91 committed:
EPMUII-10417:
- test gh workflow

rrodionov91 committed:
EPMUII-10417:
- test gh workflow

rrodionov91 committed:
EPMUII-10417:
- test gh workflow

alterfo commented on e2e-tests/playwright.config.ts:
```diff
@@ -0,0 +1,86 @@
+import { defineConfig, devices } from '@playwright/test';
+
+/**
+ * Read environment variables from file.
+ * https://github.com/motdotla/dotenv
+ */
+// import dotenv from 'dotenv';
+// import path from 'path';
+// dotenv.config({ path: path.resolve(__dirname, '.env') });
```
```suggestion
```

rrodionov91 replied:
It is default generated comment. We are not going to introduce env variables in current PR

alterfo commented on e2e-tests/playwright.config.ts:
```diff
@@ -0,0 +1,86 @@
+    // baseURL: 'http://127.0.0.1:3000',
+
+    /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
+    trace: 'off',
```
```suggestion
    trace: 'on-first-retry',
    screenshot: 'only-on-failure',
```

rrodionov91 replied:
Traces increase the report size dramatically. According to my experience they are not so useful because every report contains test steps log and it is easy to run it locally with debug option if needed (usually not often). So I'm going to keep it off in this PR

alterfo replied:
No objections

alterfo commented on .github/workflows/playwright.yml:
```diff
@@ -0,0 +1,31 @@
+    - uses: actions/checkout@v4
+    - uses: actions/setup-node@v4
+      with:
+        node-version: 16
```
```suggestion
        node-version: 18
```

alterfo commented on e2e-tests/Dockerfile-app:
```diff
@@ -0,0 +1,8 @@
+FROM node:16-bullseye-slim
```
```suggestion
FROM node:18-bullseye-slim
```

rrodionov91 committed:
EPMUII-10417:
- fixed failed test
- updated package-lock after rebase
- removed excessive operations in test running scripts

rrodionov91 committed:
EPMUII-10417:
- fixed failed test
- updated package-lock after rebase
- removed excessive operations in test running scripts
- added shutting services down after test execution
- added prepare scripts for tests

rrodionov91 committed:
EPMUII-10417:
- fixed tests build

rrodionov91 committed:
EPMUII-10417:
- added docs
- fixed test parametrization

rrodionov91 committed:
EPMUII-10417:
- fixed docs
